### PR TITLE
Feat: Add is admin utility

### LINKED ISSUE #566: Abstract out isAdmin check to new utility in /utils
Complete the following REQUIRED checkboxes:
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)

<s>-   [ ] The title of this issue follows the `command name: brief description of request` format, e.g. `/help: add optional @user parameter`</s>

The following checkbox is OPTIONAL:

-   [x] I would like to be assigned this issue to work on it

<hr>

**1. Description of the Feature Request:**
We currently duplicate the code for checking if an user is an "admin" in a number of places using roughly this snippet[1]
```js
      authorMember.roles.cache.forEach((value) => {
        if (config.roles.adminRolesName.includes(value.name)) {
          isAdmin = true;
        }
      });
```

this duplicates code, and also can be simplified to
```js
const isAdmin = member.roles.cache.some((role) => roles.adminRolesName.includes(role.name))
```

It would be beneficial to abstract this functionality out to a new `is-admin.js` file in `/utils`

[1] Taken from `points.js`

**2. Acceptance Criteria:**
- [ ] There is an utility function which checks if an member is an admin
- [ ] All code that needs to check this uses the utility function

### PR DESCRIPTION
**Pull request by Asartea**
## Because
- We currently duplicate this code

## This PR
- Adds an `isAdmin(member)` utility
- Replace all code that did this previously with usages of the new function
- Fixes using `Map()` for mocking in `points.test.js` to use `Collection()` instead
- Runs prettier over all affected files

## Issue
Closes #566

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
<s>-   [] The title of this PR follows the `location of change: brief description of change` format, e.g. `Callbacks command: Update verbiage`</s>
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If this PR adds new features or functionality, I have added new tests
-   [x] If applicable, I have ensured all tests related to any command files included in this PR pass, and/or all snapshots are up to date

### CODE REVIEW

File: events/message-create.js [hunk 1]
```diff
@@ -46,9 +48,7 @@ module.exports = {
      */
     let isAdminMessage = false;
     try {
-      isAdminMessage = message.member.roles.cache.some(
-        (r) => config.roles.adminRolesName.includes(r.name),
-      );
+      isAdminMessage = isAdmin(message.member);
     } catch (e) {
       //  The only 'con' is a command or message gets ignored.
     }
```
Comment (MaoShizhong) [inline events/message-create.js:54]: Looks like this try/catch (and the one below for isMessageAuthorNobot) is only to ignore null errors with `message.member`.

Both of these try catches can be reduced to:

```js
    const isAdminMessage = isAdmin(message.member);
    const isMessageAuthorNobot = message.member?.roles.cache.has(
      config.roles.NOBOTRoleId,
    );
```
With `isAdmin` using `?.` for `return member?.roles.cache.....`

**Review by MaoShizhong — COMMENTED** [review]
Awesome work!

Further to the two points below, we also don't actually need the `@discordjs/collection` package. `discord.js` already exports `Collection` (which comes from their utility package), so we can just destructure `Collection` from the import, then when mocking `'discord.js'`, just have `...jest.requireActual('discord.js'),` before our mocked properties.

e.g.
```js
jest.mock('discord.js', () => ({
  ...jest.requireActual('discord.js'),
  Member: jest.fn().mockImplementation((roles) => ({
    roles: {
      cache: roles,
    },
  })),
}));
```
And the equivalent for `points.test.js`

File: services/getting-hired-message.service.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ class GettingHiredMessageService {
     const userId = message.member.id;
 
     try {
-      const userIsCached = await this.#isCached(userId);
+      const userIsCached = false;
```
[the remark this answers is not in the document]
Comment (Asartea) [reply services/getting-hired-message.service.js:15]: I'm 99% sure this is a testing thing that made it in: thanks for spotting this

**Comment by Asartea** [comment]
✅ Done

**Review by MaoShizhong — COMMENTED** [review]
Very much a nit, but mind renaming any `memberMap` variables in both test files to `memberCollection`?
Once that's done, the rest looks good to me! :pray:

File: botCommands/points.test.js [hunk 3]
```diff
@@ -645,7 +615,8 @@ describe('?++ callback', () => {
 
   it('returns correct output for a user who does not have an admin role', async () => {
     const mentionedUser = User([], 2, 20);
-    const memberMap = new Map();
+    // change to use Collection
```
Comment (MaoShizhong) [inline botCommands/points.test.js:618]: Shouldn't need this comment
```suggestion
```

**Comment by MaoShizhong** [comment]
I didn't even realise I didn't formally approve. Well that's something to address.

**Pull request merged**